docs/src/pages/components/app-bar/BackToTop.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import AppBar from '@material-ui/core/AppBar';
+import Toolbar from '@material-ui/core/Toolbar';
+import Typography from '@material-ui/core/Typography';
+import { makeStyles } from '@material-ui/core/styles';
+import CssBaseline from '@material-ui/core/CssBaseline';
+import useScrollTrigger from '@material-ui/core/useScrollTrigger';
+import Box from '@material-ui/core/Box';
+import Container from '@material-ui/core/Container';
+import Fab from '@material-ui/core/Fab';
+import KeyboardArrowUpIcon from '@material-ui/icons/KeyboardArrowUp';
+import Fade from '@material-ui/core/Fade';
+
+const useStyles = makeStyles({
+  fab: {
+    position: 'fixed',
+    bottom: 16,
+    right: 16,
+  },
+});
+
+function ScrollTop(props) {
+  const { children, window } = props;
+  const classes = useStyles();
+  // Note that you normally won't need to set the window ref as useScrollTrigger
+  // will default to window.
+  // This is only being set here because the demo is in an iframe.
+  const trigger = useScrollTrigger({
+    target: window ? window() : undefined,
+    disableHysteresis: true,
+    threshold: 200,
+  });
+
+  return (
+    <Fade direction="down" in={trigger} className={classes.fab}>
+      {children}
+    </Fade>
+  );
+}
+
+ScrollTop.propTypes = {
+  children: PropTypes.element.isRequired,
+  /**
+   * Injected by the documentation to work in an iframe.
+   * You won't need it on your project.
+   */
+  window: PropTypes.func,
+};
+
+export default function BackToTop(props) {
+  const handleClick = () => {
+    console.log('click', document.querySelector('#back-to-top'));
+    document.querySelector('#back-to-top').scrollIntoView({ behavior: 'smooth' });
+  };
+
+  return (
+    <React.Fragment>
+      <CssBaseline />
+      <AppBar>
+        <Toolbar>
+          <Typography variant="h6">Scroll to Hide App Bar</Typography>
+        </Toolbar>
+      </AppBar>
+      <Toolbar id="back-to-top" />
```

Comment:
Duplicate Toolbar?

Reply:
I'm sorry, I should have made the PR a draft.

Reply:
It's motivated by https://ant.design/components/back-top/.

Reply:
> I should have made the PR a draft.

I wish that could be set as the default...

Reply:
> I wish that could be set as the default...

Or that we could change it afterward.

> Duplicate Toolbar?

The double toolbar strategy is used in the other demos too.

Reply:
Oh, right. It sets the top "padding" for the content.